src/gluonts/nursery/autogluon_tabular/predictor.py

```diff
@@ -333,3 +338,48 @@ def predict(
             return self._predict_batch(
                 dataset, batch_size=batch_size, **kwargs
             )
+
+    def serialize(self, path: Path) -> None:
+        # call Predictor.serialize() in order to serialize the class name
+
+        super().serialize(path)
+
+        # serialize self.ag_model
+        # auto gluon predictor should be saved automatically?
+        # self.ag_model.save()
```

Comment:
The autogluon model is saved already somewhere: but maybe we want to save it again under the path specified here? I understand the current mechanism: the estimator tells the prediction (when creating it) where the autogluon model is stored. However there is a point in making the “product” of the current `serialize` method here self-contained: if you serialize to path A, you can zip it and store it and it will have everything.

What do you think?

Reply:
Inside serialize, we could just move all the autogluon files to this new location. And then do:
```
ag_model = None
ag_model = TabularPredictor.load(NEW_LOCATION)
...
```

Reply:
Yes!

Reply:
I can do it in this way. But there is a problem, if I move the ag_model to NEW_LOCATION, then all the files for ag_model are in NEW_LOCATION but the path in ag_model still refer to the OLD_LOCATION where it is first created. I think this might lead to some errors.

Reply:
Currently I remove the input of path when training the estimator. And all the files for tabular will be automatically moved to the place specified when doing the serialization.